src/staticanalysis/bot/static_analysis_bot/coverity/coverity.py

```diff
@@ -0,0 +1,257 @@
+
+        logger.info('Building command files from compile_commands.json')
+
+        # Retrieve the revision files with build commands associated
+        commands_list = self.get_files_with_commands()
+        assert commands_list is not [], 'Commands List is empty'
+        logger.info('Built commands for {} files'.format(len(commands_list)))
+
+        cmd = ['gecko-env', self.cov_run_desktop, '--setup']
+        logger.info('Running Coverity Setup', cmd=cmd)
+        try:
+            subprocess.run(cmd, cwd=self.cov_path, check=True, stdout=subprocess.PIPE)
+        except subprocess.CalledProcessError as e:
+            raise AnalysisException('coverity', 'Coverity Setup failed: {}'.format(e.output))
+
+        cmd = ['gecko-env', self.cov_configure, '--clang']
+        logger.info('Running Coverity Configure', cmd=cmd)
+        try:
+            subprocess.run(cmd, cwd=self.cov_path, check=True, stdout=subprocess.PIPE)
+        except subprocess.CalledProcessError as e:
+            raise AnalysisException('coverity', 'Coverity Configure failed: {}'.format(e.output))
+
+        # For each element in commands_list run `cov-translate`
+        for element in commands_list:
+            cmd = [
+                'gecko-env', self.cov_translate, '--dir', self.cov_idir_path,
+                element['command']
+            ]
+            logger.info('Running Coverity Tranlate', cmd=cmd)
+            try:
+                subprocess.run(cmd, cwd=element['directory'], check=True, stdout=subprocess.PIPE)
+            except subprocess.CalledProcessError as e:
+                raise AnalysisException('coverity', 'Coverity Translate failed: {}'.format(e.output))
+
+        # Once the capture is performed we need to do the actual Coverity Desktop analysis
+        cmd = [
+            'gecko-env', self.cov_run_desktop, '--json-output-v6',
+            'cov-results.json', '--strip-path', settings.repo_dir
+        ]
+        for element in commands_list:
+            cmd.append(element['file'])
+        logger.info('Running Coverity Analysis', cmd=cmd)
+        try:
+            subprocess.run(cmd, cwd=self.cov_state_path, check=True, stdout=subprocess.PIPE)
+        except subprocess.CalledProcessError as e:
+            raise AnalysisException('coverity', 'Coverity Analysis failed: {}'.format(e.output))
+
+        # Write the results.json to the artifact directory to have it later on for debug
+        coverity_results_path = os.path.join(self.cov_state_path,
+                                             'cov-results.json')
+        coverity_results_path_on_tc = os.path.join(
+            settings.taskcluster.results_dir, 'cov-results.json')
+        import shutil
+        shutil.copyfile(coverity_results_path, coverity_results_path_on_tc)
+
+        # Parsing the issues from coverity_results_path
+        logger.info('Parsing Coverity issues')
+        return self.return_issues(coverity_results_path, revision)
```

Comment:
Not sure using a function is useful or expressive here. Did you do that for unit tests ?

Reply:
Yes this is the reasoning behind this to have some sort of pseudo-unit tests even though we cannot have something that reliable as we have for clang-tidy and clang-format.

Reply:
Ok ! Can you implement this one in the PR ?

Reply:
Sure i will do the unit tests last.